core/src/test/java/ai/timefold/solver/core/impl/move/streams/maybeapi/provider/ChangeMoveProviderTest.java

```diff
@@ -79,8 +81,67 @@ void fromSolution() {
         });
     }
 
-    private MoveStreamSession<TestdataSolution> createSession(DefaultMoveStreamFactory<TestdataSolution> moveStreamFactory,
-            SolutionDescriptor<TestdataSolution> solutionDescriptor, TestdataSolution solution) {
+    @Test
+    void fromSolutionBasicVariableAllowsUnassigned() {
+        var solutionDescriptor = TestdataAllowsUnassignedSolution.buildSolutionDescriptor();
+        var variableMetaModel = solutionDescriptor.getMetaModel()
+                .entity(TestdataAllowsUnassignedEntity.class)
+                .genuineVariable()
+                .ensurePlanningVariable();
+        var moveStreamFactory = new DefaultMoveStreamFactory<>(solutionDescriptor);
+        var moveProvider = new ChangeMoveProvider<>(variableMetaModel);
+        var moveProducer = moveProvider.apply(moveStreamFactory);
+
+        var solution = TestdataAllowsUnassignedSolution.generateSolution(2, 2);
+        var firstEntity = solution.getEntityList().get(0);
+        var secondEntity = solution.getEntityList().get(1);
+        var firstValue = solution.getValueList().get(0);
+        var secondValue = solution.getValueList().get(1);
+        var moveStreamSession = createSession(moveStreamFactory, solutionDescriptor, solution);
+
+        var moveIterable = moveProducer.getMoveIterable(moveStreamSession);
+        assertThat(moveIterable).hasSize(4);
+
+        var moveList = StreamSupport.stream(moveIterable.spliterator(), false)
+                .map(m -> (ChangeMove<TestdataAllowsUnassignedSolution, TestdataAllowsUnassignedEntity, TestdataValue>) m)
+                .toList();
+        assertThat(moveList).hasSize(4);
+
+        var firstMove = moveList.get(0);
+        assertSoftly(softly -> {
+            softly.assertThat(firstMove.extractPlanningEntities())
+                    .containsExactly(firstEntity);
+            softly.assertThat(firstMove.extractPlanningValues())
+                    .containsExactly(firstValue);
+        });
+
+        var secondMove = moveList.get(1);
+        assertSoftly(softly -> {
+            softly.assertThat(secondMove.extractPlanningEntities())
+                    .containsExactly(firstEntity);
+            softly.assertThat(secondMove.extractPlanningValues())
+                    .containsExactly(secondValue);
+        });
+
+        var thirdMove = moveList.get(2);
```

Comment:
Please provide a comment on this test. I needed to run it to realize that entity 1 holds the value 1.